### PR TITLE
GL2PS

### PR DESCRIPTION
**Pull request by henryiii**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@chrisburr, I've started the recipe you wanted here, but I won't have time to work on it for a while. I think we mostly need a test (and then to fix whatever is missing if the test fails). I'll make sure you have permission to push if you want to do anything with it.

### CODE REVIEW

**Comment by henryiii** [comment]
We get a lot of warnings like this one:

```
 'glutBitmapCharacter' is deprecated: first deprecated in macOS 10.9 - OpenGL API deprecated. (Define
      GL_SILENCE_DEPRECATION to silence these warnings) [-Wdeprecated-declarations]
    glutBitmapCharacter(GLUT_BITMAP_HELVETICA_12, string[i]);
```

Is there a way to use a conda-forge version, perhaps?

**Comment by henryiii** [comment]
I think the problem is here:

```
if(OPENGL_FOUND)
  add_library(lib STATIC gl2ps.c gl2ps.h)
  set_target_properties(lib PROPERTIES OUTPUT_NAME gl2ps)

  add_library(shared SHARED gl2ps.c gl2ps.h)
  target_link_libraries(shared ${EXTERNAL_LIBRARIES})
  set_target_properties(shared PROPERTIES OUTPUT_NAME gl2ps
    VERSION ${GL2PS_MAJOR_VERSION}.${GL2PS_MINOR_VERSION}.${GL2PS_PATCH_VERSION}
    SOVERSION ${GL2PS_MAJOR_VERSION}.${GL2PS_MINOR_VERSION})
```

On Windows, both produce `gl2ps.lib`.

**Comment by henryiii** [comment]
This may not support anything beyond mingw on Windows. If that's the case, we could disable Windows here then add built-in gl2ps to ROOT on Windows if/when it gains proper Windows support.

**Comment by chrisburr** [comment]
@conda-forge/staged-recipes Ready for review!

File: recipes/gl2ps/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+# Enable bash strict mode
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+
+mkdir -p build
+cd build
+
+cmake -LAH \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DCMAKE_BUILD_TYPE=${CMAKE_PLATFORM_FLAGS[@]+"${CMAKE_PLATFORM_FLAGS[@]}"} \
```
Comment (isuruf) [inline recipes/gl2ps/build.sh]: Who is setting `CMAKE_PLATFORM_FLAGS` ?

**Comment by chrisburr** [comment]
@conda-forge-admin, please relint

**Comment by scopatz** [comment]
This LGTM!

File: recipes/gl2ps/build.sh (hunk 1 above)
Comment (henryiii) [reply recipes/gl2ps/build.sh]: I got this from HepMC3:

https://github.com/conda-forge/hepmc3-feedstock/blob/c9c43d086ad39bbdfe1a1cbda5f7a6fc82793a50/recipe/build.sh#L14

Which I expect was (wrongly) copied from [root](https://github.com/conda-forge/root-feedstock/blob/master/recipe/build_root-base.sh), which actually does set this.

**Comment by chrisburr** [comment]
Thanks for the review @scopatz and @isuruf! 🍰

**Pull request merged**